From c44aac2e3e45f9a3a36ba80704104082e5a71100 Mon Sep 17 00:00:00 2001
From: Vaibhav Hiremath <hvaibhav@ti.com>
Date: Mon, 20 Jul 2009 22:17:52 +0530
Subject: [PATCH] OMAP3EVM: Build issue fixed.

The x-loader changes was broken for OMAP3EVM, fixed -

	- GPMC macros were not under proper board macro
	- Warning fixed from mmc.c
---
 cpu/omap3/mmc.c               |    3 ++-
 include/asm/arch-omap3/mem.h  |    6 +++---
 include/configs/omap3517evm.h |    2 +-
 3 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/cpu/omap3/mmc.c b/cpu/omap3/mmc.c
index b9dd5d0..2453477 100755
--- a/cpu/omap3/mmc.c
+++ b/cpu/omap3/mmc.c
@@ -46,8 +46,9 @@ block_dev_desc_t *mmc_get_dev(int dev)
 
 unsigned char mmc_board_init(void)
 {
-	unsigned int value = 0;
 #if defined (CONFIG_OMAP34XX)
+	unsigned int value = 0;
+
 	value = CONTROL_PBIAS_LITE;
 	CONTROL_PBIAS_LITE = value | (1 << 2) | (1 << 1) | (1 << 9);
 
diff --git a/include/asm/arch-omap3/mem.h b/include/asm/arch-omap3/mem.h
index 8a92484..ae73a9f 100644
--- a/include/asm/arch-omap3/mem.h
+++ b/include/asm/arch-omap3/mem.h
@@ -324,7 +324,7 @@ typedef enum {
 # define SMNAND_GPMC_CONFIG5 0x010C1414
 # define SMNAND_GPMC_CONFIG6 0x00000A80
 
-#if defined (CONFIG_OMAP34xx)
+#if defined (CONFIG_OMAP34XX)
 
 # define M_NAND_GPMC_CONFIG1 0x00001800
 # define M_NAND_GPMC_CONFIG2 SMNAND_GPMC_CONFIG2
@@ -372,7 +372,7 @@ typedef enum {
 # define SMNAND_GPMC_CONFIG6 0x00000A80
 # define SMNAND_GPMC_CONFIG7 0x00000C44
 
-#if defined (CONFIG_OMAP34xx)
+#if defined (CONFIG_OMAP34XX)
 
 # define M_NAND_GPMC_CONFIG1 0x00001800
 # define M_NAND_GPMC_CONFIG2 SMNAND_GPMC_CONFIG2
@@ -445,7 +445,7 @@ typedef enum {
 # define SMNAND_GPMC_CONFIG6 0x1F0F0A80
 # define SMNAND_GPMC_CONFIG7 0x00000C44
 
-#if defined (CONFIG_OMAP34xx)
+#if defined (CONFIG_OMAP34XX)
 
 # define M_NAND_GPMC_CONFIG1 0x00001800
 # define M_NAND_GPMC_CONFIG2 SMNAND_GPMC_CONFIG2
diff --git a/include/configs/omap3517evm.h b/include/configs/omap3517evm.h
index 95deacb..16dc8d8 100644
--- a/include/configs/omap3517evm.h
+++ b/include/configs/omap3517evm.h
@@ -124,7 +124,7 @@
 
 #define GPMC_CONFIG              (OMAP34XX_GPMC_BASE+0x50)
 
-#if defined (CONFIG_OMAP34xx) || (CONFIG_OMAP3517EVM)
+#if defined (CONFIG_OMAP3517EVM)
 #define GPMC_NAND_COMMAND_0      (OMAP34XX_GPMC_BASE+0x7C)
 #define GPMC_NAND_ADDRESS_0      (OMAP34XX_GPMC_BASE+0x80)
 #define GPMC_NAND_DATA_0         (OMAP34XX_GPMC_BASE+0x84)
-- 
1.5.4.3

